Update Deprecations rules to latest RuboCop format

scottvidmar opened the pull request:
## Description
This refactors deprecations rules to use the latest format in use by RuboCop:
https://docs.rubocop.org/rubocop/development.html

No new functionality added.
## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:
- [X ] I have read the **CONTRIBUTING** document.
- [ X] I have run the pre-merge tests locally and they pass.
- [ X] I have updated the documentation accordingly.
- [X ] I have added tests to cover my changes.
- [X ] All new and existing tests passed
- [ X] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

## Testing
Ran existing RSpec tests, as this is not adding or removing functionality.

$ bundle exec rspec spec -f p
Run options: include {:focus=>true}

All examples were filtered out; ignoring {:focus=>true}

Randomized with seed 304
................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................

Finished in 0.88479 seconds (files took 2.08 seconds to load)
752 examples, 0 failures

Randomized with seed 304

tas50 commented on lib/rubocop/cop/chef/deprecation/chef_handler_recipe.rb:
```diff
@@ -37,15 +37,12 @@ class ChefHandlerRecipe < Cop
             (send nil? :include_recipe (str {"chef_handler" "chef_handler::default"}))
           PATTERN
 
+          extend AutoCorrector
```
I'd prefer to have the extends right up with the class so it's clear that this class extends AutoCorrector

scottvidmar replied:
Agreed. Fixed.

tas50 commented on lib/rubocop/cop/chef/deprecation/chef_rewind.rb:
```diff
@@ -61,47 +61,54 @@ class ChefRewind < Cop
-                return
-              end
+          def fixer(node, corrector)
+            rewind_gem_install?(node) do
```
We should be able to extract this mess into each corrector not. It's one of the reasons they did the new autocorrect inline format.

scottvidmar replied:
Fixed

tas50 commented on lib/rubocop/cop/chef/deprecation/cheffile.rb:
```diff
@@ -22,18 +22,19 @@ module Chef
-          def investigate(processed_source)
-            return if processed_source.blank?
+          def on_new_investigation
+            f_name = processed_source.file_path
```
We limit the filename in the config so there's no need to do the file name check again.

scottvidmar replied:
Fixed.

tas50 replied:
@scottvidmar This doesn't look fixed yet

scottvidmar replied:
Now it should be fixed.

tas50 commented on lib/rubocop/cop/chef/deprecation/chocolatey_package_uninstall_action.rb:
```diff
@@ -34,24 +34,21 @@ module ChefDeprecations
-                add_offense(action, location: :expression, message: MSG, severity: :warning) if action.source == ':uninstall'
+                add_offense(action, message: MSG, severity: :warning) do |corrector|
+                  corrector.replace(action, ':remove')
+                end if action.source == ':uninstall'
```
It's a lot easier to follow these by removing the ```if``` on the block at line 48 and instead add ```next unless action.source == ':uninstall' ``` before the ```add_offense```

scottvidmar replied:
These should be addressed.

tas50 commented on lib/rubocop/cop/chef/deprecation/deprecated_mixins.rb:
```diff
@@ -51,24 +51,29 @@ class UsesDeprecatedMixins < Cop
             deprecated_mixin?(node) do
-              add_offense(node, location: :expression, message: MSG, severity: :warning)
+              add_offense(node, message: MSG, severity: :warning) do |corrector|
+                fixer(node, corrector)
```
We should be able to call the individual autocorrects here as well

scottvidmar replied:
Should be all fixed up.

scottvidmar commented:
Addressed PR feedback. Tests still passing:

 bundle exec rspec spec -f p
Run options: include {:focus=>true}

All examples were filtered out; ignoring {:focus=>true}

Randomized with seed 1869
................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................

Finished in 1.3 seconds (files took 1.12 seconds to load)
752 examples, 0 failures

Randomized with seed 1869

tas50 commented on lib/rubocop/cop/chef/deprecation/launchd_deprecated_hash_property.rb:
```diff
@@ -33,23 +33,20 @@ module ChefDeprecations
-          def autocorrect(node)
-            lambda do |corrector|
-              corrector.replace(node.loc.expression, node.loc.expression.source.gsub(/^hash/, 'plist_hash'))
+            return unless match_property_in_resource?(:launchd, 'hash', node) do |offense|
```
There shouldn't be a need for the return unless bit here. If it matches it'll run the block

scottvidmar replied:
Good call. Removed that bit.

tas50 commented:
Thanks for taking the time to fix these all up. It'll allow us to upgrade RuboCop whenever they rip out the old classes.
